### PR TITLE
Openemr fhir smart bulk

### PR DESCRIPTION
**Pull request by adunsulag**
Hooray for check marks!  This impelements the [Bulk FHIR or Flat FHIR Bulk export for bulk fhir export.](https://hl7.org/fhir/uv/bulkdata/)  
image

Currently it relies on the fact that OpenEMR has a single group identified by the group number '1'.  If we find out we have to actually support different kinds of patient populations ie groups we'll have to revisit this.

I removed the system write permissions as we don't support via FHIR a bulk import at this point in time.  The spec from HL7 is still being tested out on what they want to do there.

I added some additional fixes to enforce access rights on document search, removing some phpseclib notices, fixed the Data Absent Reason for blood pressure (which also fixes an issue on the single patient api) and implemented the list endpoint for Provenance.

I do still have a warning notice (passes spec, but may fail certification) on MedicationRequest's requestor.  For some reason it doesn't like the DAR extension if the medication's requestor (person that created the medication issue) is missing.

### CODE REVIEW

File: src/Services/FHIR/Observation/FhirObservationVitalsService.php [hunk 1]
```diff
@@ -664,6 +664,12 @@ private function columnHasPositiveFloatValue($column, $record)
 
     private function populateBloodPressurePanel(FHIRObservation $observation, $dataRecord)
     {
+            // Based on conversations with Jerry Padget and Brady Miller on August 14th 2021 we decided that if the values
+            // were both 0 for bpd and bps we would treat this as a data absent reason.  In this case an attempt was made
+            // to record the data but no value was recorded (such as the blood pressure cuff becoming loose).
+        if ($dataRecord['bpd'] == 0 && $dataRecord['bps'] == 0) {
+            $observation->setDataAbsentReason(UtilsService::createDataAbsentUnknownCodeableConcept());
```
Comment (bradymiller) [inline src/Services/FHIR/Observation/FhirObservationVitalsService.php:671]: yes!! the 0 hack!
2DV-2

Comment (adunsulag) [reply src/Services/FHIR/Observation/FhirObservationVitalsService.php:671]: I'm going to have to find a sad face emoji.

**Comment by adunsulag** [comment]
So both the openemr pr and openemr pf command is not showing me the style problems... so annoying.

**Comment by bradymiller** [comment]
github actions can smell fear :)

https://github.com/openemr/openemr/issues/4073

**Comment by bradymiller** [comment]
code looks good to me and feel free to bring in after appease the github actions styling god :)
thanks @adunsulag !

**Comment by stephenwaite** [comment]
that's odd @adunsulag, `openemr-cmd pf` fixes this error after checking out branch

**Pull request merged**